pyartcd/pyartcd/pipelines/brew_scan_osh.py

```diff
@@ -0,0 +1,111 @@
+"""
+[STORES STATE IN WORKSPACE]
+This job scans the candidate tags for a particular version, and triggers scans for builds that are tagged into it.
+"""
+import click
+import json
+import base64
+import yaml
+import os
+from ghapi.all import GhApi
+from pyartcd import exectools
+from pyartcd.runtime import Runtime
+from pyartcd.cli import cli, pass_runtime, click_coroutine
+
+FILE_PATH = f"{os.getenv('WORKSPACE')}/event.json"
+
+
+class OshScan:
+    def __init__(self, runtime: Runtime, email: str, version: str):
+        self.runtime = runtime
+        self.email = email
+        self.version = version
+        self.major, self.minor = self.version.split(".")
+        self.last_event = {}
+        self.tags = None
+
+    async def _get_ocp_candidate_tags(self):
+        """
+        Get the candidate tags activated for a specific OCP version
+        """
+        # GITHUB_TOKEN env variable needs to be set in the Jenkinsfile to avoid rate limiting
+        api = GhApi(owner="openshift-eng", repo="ocp-build-data")
+        branch = f"openshift-{self.version}"
+        file_path = "erratatool.yml"
+
+        response = api.repos.get_content(
+            path=file_path,
+            ref=branch,
+        )
+        # Decode the base64 content
+        content = response["content"]
+        decoded_content = base64.b64decode(content).decode("utf-8")
+        yaml_data = yaml.safe_load(decoded_content)
+
+        return yaml_data["brew_tag_product_version_mapping"]
+
+    async def _set_candidate_tags(self):
+        data = await self._get_ocp_candidate_tags()
+
+        self.tags = [tag.replace("{MAJOR}", self.major).replace("{MINOR}", self.minor) for tag in
+                     data]
+
+    def _get_last_event(self):
```

Comment:
How about using redis instead to store and retrieve?

Reply:
+1. Let's think about a pattern to keep data organized. We currently have `count` and `failures` as main keys. We could add something like `appdata:{app_name}:{key}` where appname would be `sast_scan` in this case.